frame/support/src/inherent.rs

```diff
@@ -36,6 +39,11 @@ pub trait ProvideInherent {
 	const INHERENT_IDENTIFIER: self::InherentIdentifier;
 
 	/// Create an inherent out of the given `InherentData`.
+	///
+	/// NOTE: All checks necessary to ensure that the inherent is correct and that can be done in
+	/// the runtime should happen in the returned `Call`.
+	/// E.g. if this provides a block producer, check in the returned extrinsic that the producer is
+	/// eligible to create the block.
```

Comment:
I don't very like this example, probably for the same reason as said above by Basti.

Like currently the block producer is passed by using digest, but I guess we could use an inherent.
Then the inherent could just write the author into a storage (which can be removed in on_finalize in order not to get into the block).
Later other pallets could check that the author is able to produce a block, and just panic in case he is not able to produce a block.

Thus the check whether the producer is eligible to create the block will not happen in the call, but later in the block production.

I think what is written above:
> /// NOTE: All checks necessary to ensure that the inherent is correct and that can be done in
	/// the runtime should happen in the returned `Call`.
	
Is good, but maybe as an example I would write:
> E.g. if this provide the timestamp, the call will check that the given timestamp is increasing the old timestamp by more than a minimum and it will also check that the timestamp hasn't already been set.
> Later in the `on_finalize` hook, the pallet could also ensure that the timestamp is set.